02-peer-protocol.md

```diff
@@ -914,21 +978,20 @@ This message initiates the v2 channel establishment workflow.
    * [`point`:`delayed_payment_basepoint`]
    * [`point`:`htlc_basepoint`]
    * [`point`:`first_per_commitment_point`]
-   * [`byte`:`channel_flags`]
+   * [`bigsize`:`channel_flags_len`]
```

Comment:
This is a departure from how `channel_flags` currently work in v1. Is this in line with the channel_types expansion?

Reply:
It's true that it differs from v1 `channel_flags`, but my hope is that we will have switched to always using dual funding before we fill the 8 bits of the current `channel_flags` (so in practice they will be compatible). And then with that change, we won't be constrained to only 8 potential channel flags in the future (even if we don't need it, why not have it just in case, it doesn't hurt?)!

I think the channel flags are orthogonal to channel types, they control "smaller" details of the channel (like whether the channel is public or private). I could be wrong though, what we do in the future with `channel_type`s and `channel_flag`s can turn out to be very different from how I see it, but at least we're keeping options open by having a variable-length field here.